src/main/java/uk/gov/companieshouse/authcodenotification/controller/AuthCodeNotificationController.java

```diff
@@ -63,6 +64,8 @@ public ResponseEntity<Object> sendEmail(@RequestHeader(value = ERIC_REQUEST_ID_K
 
         try {
             authCodeNotificationService.sendAuthCodeEmail(requestId, authCode, companyNumber);
+        } catch (EntityNotFoundException nfe) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
We should be logging the exception details at this point. In the catch block below, too.

Reply:
Added logging details.

Reply:
I would replace the null with the exception otherwise we loose the original stack trace of the cause

Reply:
plus, i think the not found error should be Info instead of errorContext, in which case we wouldn't pass the exception in to the logger. @mwestacott @pjames-ch do you think the 404 should be treated as normal operation and info level or error level?

Reply:
I'll wait to see what people say but it does seem like an error as it would prevent the email from being sent.

Reply:
I think this could be logged at info level as this controller code is making the decision that this isn't really an error (at least not a totally unexpected one), it's simply that the resource isn't found. But... if logged at info level, can the stack trace still be output?

Reply:
Changed to info

Reply:
I don't think the logger has options for logging stacktrace at info level, you'd have to add it as part of the message or logmap.

Reply:
I think so. Didn't resolve, only so that @markpit could read the comments and check he's happy.